src/realm/db.cpp

```diff
@@ -379,7 +381,17 @@ class Ringbuffer {
     // IMPORTANT II:
     // To ensure proper alignment across all platforms, the SharedInfo structure
     // should NOT have a stricter alignment requirement than the ReadCount structure.
-    ReadCount data[init_readers_size];
+    ReadCount m_data[init_readers_size];
+
+    // Silence UBSan errors about out-of-bounds reads on m_data by casting to a pointer
+    ReadCount* data() noexcept
+    {
+        return m_data;
+    }
+    const ReadCount* data() const noexcept
+    {
+        return m_data;
+    }
```

Comment:
The flexible array member idiom is not actually legal C++, and UBSan complains about out-of-bounds reads once the ring buffer expands beyond the starting size. In practice all compilers allow it, so this is just hiding it from UBSan.